compiler/next/include/chpl/uast/UseClause.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_USECLAUSE_H
+#define CHPL_UAST_USECLAUSE_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Decl.h"
+#include "chpl/uast/Expression.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents an individual use clause. For example:
+
+  \rst
+  .. code-block:: chapel
+    // This contains two use clauses, 'Foo as X' and 'Baz as Y'.
+    use Foo as X, Baz as Y;
+
+  \endrst
+ */
+class UseClause final : public Expression {
+ public:
+  enum LimitationClauseKind {
+    EXCEPT,
+    ONLY,
+    NONE
+  };
+
+ private:
+  UseClause(ASTList children,  LimitationClauseKind limitationClauseKind,
+            int8_t limitationClauseChildNum,
+            int numLimitations)
+    : Expression(asttags::UseClause, std::move(children)),
+      limitationClauseKind_(limitationClauseKind),
+      limitationClauseChildNum_(limitationClauseChildNum),
+      numLimitations_(numLimitations) {
+
+    switch (limitationClauseKind_) {
+      case EXCEPT: assert(numLimitations >= 1); break;
+      case NONE: assert(numLimitations_ == 0); break;
+      case ONLY: assert(numLimitations_ >= 0); break;
+    }
+  }
+
+  // No need to check 'nameChildNum_'.
+  bool contentsMatchInner(const ASTNode* other) const override {
+    const UseClause* rhs = other->toUseClause();
+    return this->limitationClauseKind_ == rhs->limitationClauseKind_ &&
+      this->limitationClauseChildNum_ == rhs->limitationClauseChildNum_ &&
+      this->numLimitations_ == rhs->numLimitations_ &&
+      this->expressionContentsMatchInner(rhs);
+  }
+
+  void markUniqueStringsInner(Context* context) const override {
+    expressionMarkUniqueStringsInner(context);
+  }
+
+  // This always exists and its position never changes.
+  static const int8_t nameChildNum_ = 0;
+
+  LimitationClauseKind limitationClauseKind_;
+  int8_t limitationClauseChildNum_;
```

Comment:
Isn't this always 1 ? Seems like you could just check `numLimitations > 0` and treat this one as 1 in that event.